packages/edit-site/src/components/layout/router.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * WordPress dependencies
+ */
+import { privateApis as routerPrivateApis } from '@wordpress/router';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+import { useIsSiteEditorLoading } from './hooks';
+import Editor from '../editor';
+import DataviewsPatterns from '../page-patterns/dataviews-patterns';
+import PagePages from '../page-pages';
+import PagePatterns from '../page-patterns';
+import PageTemplateParts from '../page-template-parts';
+import PageTemplates from '../page-templates';
+
+const { useLocation } = unlock( routerPrivateApis );
+
+export default function useLayoutAreas() {
+	const isSiteEditorLoading = useIsSiteEditorLoading();
+	const { params } = useLocation();
+	const { postType, postId, path, type, isCustom } = params ?? {};
+
+	// Regular page
+	if ( path === '/page' ) {
+		const isListLayout =
+			isCustom !== 'true' && ( ! type || type === 'list' );
```

Comment:
This is the difficulty here, basically the route definition relies on the "type" of the dataview (layout), which means I had to add this information to the URL. 

There's maybe another way to solve this but in trunk the "router" can't access this information because it's only kept in memory.

Reply:
Why do you have concerns about having the `type` as a URL param?

This makes me consider a more general point: why routing can't act as a pure function of the URL? Given a URL, I'm able to share it with other person who will see the same screen I'm seeing. This, of course, extends to things other than the view type (filters/search, pagination, etc.).

Reply:
I think the unknown here is that most of the views could either become entities in the future or be persisted in preferences or things like that, which means the "type" in the url redundant and something we'd want to ditch.

Reply:
Are we okay with showing the front page when we have no selection in `pages list`? Personally I think this a bit confusing..

Reply:
> Are we okay with showing the front page when we have no selection in pages list? Personally I think this a bit confusing..

The way I see it, it's just how the (rest of the) site editor works, so it makes sense that pages does the same.

Reply:
I agree that it's not entirely clear whether the "home page" is the right choice there. A placeholder (select item kind of thing) could work like emails but I agree that for now it's probably fine. I think it would be good to show bulk actions there somehow in case of multi-selections... But all of these can be seen as follow-up improvements.